### PR TITLE
Issue #44 create CarEntity & enum CarStatus

### LINKED ISSUE #44: Create CarEntity & enum CarStatus

### PR DESCRIPTION
**Pull request by JustRevolt**

### CODE REVIEW

File: src/main/java/com/ifmo/epampractice/entity/CarEntity.java [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+
+        CarEntity carEntity = (CarEntity) o;
+
+        return (
+                id == carEntity.id &&
+                modelId == carEntity.modelId &&
+                color.equals(carEntity.color) &&
+                regNumber.equals(carEntity.regNumber) &&
+                VIN.equals(carEntity.VIN) &&
+                status == carEntity.status
```
Comment (nikaalin) [inline src/main/java/com/ifmo/epampractice/entity/CarEntity.java]: `status == carEntity.status` 
лучше сделать через Objects.equals()

Comment (JustRevolt) [reply src/main/java/com/ifmo/epampractice/entity/CarEntity.java]: Done

**Review by nikaalin — APPROVED** [review]

**Review by a-deriabin — APPROVED** [review]

**Review by ChirkinaTanya — APPROVED** [review]

File: src/main/java/com/ifmo/epampractice/entity/CarEntity.java [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+import com.ifmo.epampractice.enums.CarStatus;
+
+import java.util.Objects;
+
+public class CarEntity {
+    private int         id;
+    private int         modelId;
+    private String      color;
+    private String      regNumber;
+    private String      VIN;
```
Comment (PavelEzhkov) [inline src/main/java/com/ifmo/epampractice/entity/CarEntity.java]: is it constant? `vin` will be ok

File: src/main/java/com/ifmo/epampractice/entity/CarEntity.java [hunk 3]
```diff
@@ -0,0 +1,100 @@
...
+    private CarStatus   status;
+
+    public CarEntity(int id, int modelId, String color, String regNumber,
+                     String VIN, CarStatus status) {
```
Comment (PavelEzhkov) [inline src/main/java/com/ifmo/epampractice/entity/CarEntity.java]: `vin`

File: src/main/java/com/ifmo/epampractice/entity/CarEntity.java [hunk 4]
```diff
@@ -0,0 +1,100 @@
...
+
+    public void setRegNumber(String regNumber) {
+        this.regNumber = regNumber;
+    }
+
+    public String getVIN() {
+        return VIN;
+    }
+
+    public void setVIN(String VIN) {
```
Comment (PavelEzhkov) [inline src/main/java/com/ifmo/epampractice/entity/CarEntity.java]: `vin`

File: src/main/java/com/ifmo/epampractice/entity/CarEntity.java [hunk 5]
```diff
@@ -0,0 +1,100 @@
...
+            return true;
+        }
+        if (o == null || getClass() != o.getClass()) {
+            return false;
+        }
+
+        CarEntity that = (CarEntity) o;
+
+        return (
+                Objects.equals(id, that.id) &&
```
Comment (PavelEzhkov) [inline src/main/java/com/ifmo/epampractice/entity/CarEntity.java]: Is int object in java? Use == for basic types  and Objects.equals() for Objects

File: src/main/java/com/ifmo/epampractice/entity/CarEntity.java [hunk 6]
```diff
@@ -0,0 +1,100 @@
...
+                Objects.equals(modelId, that.modelId) &&
```
Comment (PavelEzhkov) [inline src/main/java/com/ifmo/epampractice/entity/CarEntity.java]: ==

File: src/main/java/com/ifmo/epampractice/entity/CarEntity.java [hunk 7]
```diff
@@ -0,0 +1,100 @@
...
+                Objects.equals(color, that.color) &&
+                Objects.equals(regNumber, that.regNumber) &&
+                Objects.equals(VIN, that.VIN) &&
+                Objects.equals(status, that.status)
```
Comment (PavelEzhkov) [inline src/main/java/com/ifmo/epampractice/entity/CarEntity.java]: ==

**Review by PavelEzhkov — COMMENTED** [review]
fix it

File: src/main/java/com/ifmo/epampractice/entity/CarEntity.java (hunk 2 above)
Comment (JustRevolt) [reply src/main/java/com/ifmo/epampractice/entity/CarEntity.java]: @PavelEzhkov done

File: src/main/java/com/ifmo/epampractice/entity/CarEntity.java (hunk 3 above)
Comment (JustRevolt) [reply src/main/java/com/ifmo/epampractice/entity/CarEntity.java]: @PavelEzhkov done

File: src/main/java/com/ifmo/epampractice/entity/CarEntity.java (hunk 4 above)
Comment (JustRevolt) [reply src/main/java/com/ifmo/epampractice/entity/CarEntity.java]: @PavelEzhkov done

File: src/main/java/com/ifmo/epampractice/entity/CarEntity.java (hunk 5 above)
Comment (JustRevolt) [reply src/main/java/com/ifmo/epampractice/entity/CarEntity.java]: @PavelEzhkov done

File: src/main/java/com/ifmo/epampractice/entity/CarEntity.java (hunk 6 above)
Comment (JustRevolt) [reply src/main/java/com/ifmo/epampractice/entity/CarEntity.java]: @PavelEzhkov done

File: src/main/java/com/ifmo/epampractice/entity/CarEntity.java (hunk 7 above)
Comment (JustRevolt) [reply src/main/java/com/ifmo/epampractice/entity/CarEntity.java]: @PavelEzhkov why we need to use == to compare enums Object?

Comment (JustRevolt) [reply src/main/java/com/ifmo/epampractice/entity/CarEntity.java]: @PavelEzhkov done

**Review by PavelEzhkov — APPROVED** [review]

**Pull request merged**